java/Mekhmet-Efe_Adzhar/src/com/zoolatech/lecture4/tasks/_4/Number4.java

```diff
@@ -22,110 +22,58 @@ enum Weight {
-            return ((mass * 2.205) + " " + POUND);
-        } else if (sourceUnit == KILOGRAM && convertTo == TON) {
-            System.out.println("Converting Kilograms to OUNCE...");
-            return ((mass / 1000) + " " + POUND);
-
-            //MARK: OUNCE
-        } else if (sourceUnit == OUNCE && convertTo == GRAM) {
-            System.out.println("Converting Kilograms to OUNCE...");
-            return ((mass * 28.35) + " " + GRAM);
-        } else if (sourceUnit == OUNCE && convertTo == KILOGRAM) {
-            System.out.println("Converting Kilograms to OUNCE...");
-            return ((mass / 35.274) + " " + KILOGRAM);
-        } else if (sourceUnit == OUNCE && convertTo == POUND) {
-            System.out.println("Converting Kilograms to OUNCE...");
-            return ((mass / 16) + " " + POUND);
-        } else if (sourceUnit == OUNCE && convertTo == TON) {
-            System.out.println("Converting Kilograms to OUNCE...");
-            return ((mass / 283495231250000F) + " " + TON);
-
-            //MARK: POUND
-        } else if (sourceUnit == POUND && convertTo == GRAM) {
-            System.out.println("Converting Kilograms to OUNCE...");
-            return ((mass * 453.6) + " " + GRAM);
-        } else if (sourceUnit == POUND && convertTo == KILOGRAM) {
-            System.out.println("Converting Kilograms to OUNCE...");
-            return ((mass / 2.205) + " " + KILOGRAM);
-        } else if (sourceUnit == POUND && convertTo == OUNCE) {
-            System.out.println("Converting Kilograms to OUNCE...");
-            return ((mass * 16) + " " + OUNCE);
-        } else if (sourceUnit == POUND && convertTo == TON) {
-            System.out.println("Converting Kilograms to OUNCE...");
-            return ((mass / 2205) + " " + TON);
-
-            //MARK: TON
-        } else if (sourceUnit == TON && convertTo == GRAM) {
-            System.out.println("Converting Kilograms to OUNCE...");
-            return ((mass * 1000000) + " " + GRAM);
-        } else if (sourceUnit == TON && convertTo == KILOGRAM) {
-            System.out.println("Converting Kilograms to OUNCE...");
-            return ((mass * 1000) + " " + KILOGRAM);
-        } else if (sourceUnit == TON && convertTo == OUNCE) {
-            System.out.println("Converting Kilograms to OUNCE...");
-            return ((mass * 35270) + " " + OUNCE);
-        } else if (sourceUnit == TON && convertTo == POUND) {
-            System.out.println("Converting Kilograms to OUNCE...");
-            return ((mass * 2205) + " " + POUND);
-        }
-        return "ERROR";
-    }
-
-    static Weight enumValueByAString(String string) {
-        switch (string) {
-            case "GRAM", "Gram", "gram", "gr" -> {
-                return GRAM;
+    public float convertValue(float mass, Weight sourceUnit) {
+        switch (sourceUnit) {
+            case GRAM -> {
+                System.out.println("Converting Gram to Kilogram...");
+                return mass / 1000;
```

Comment:
how do you know the target unit is, for example, kilogram? What if it's the `pound`?